### ISSUE TITLE
HaGeZi's Allowlist Referral -- it is an allowlist but it's under blocklist

### ISSUE THREAD

**Issue by timkgh** [AdguardTeam/AdGuardHome#6207]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Darwin (aka macOS), AMD64 (aka x86_64)

### Installation

GitHub releases or script from README

### Setup

On one machine

### AdGuard Home version

v0.108.0-b.46

### Action

none

### Expected result

HaGeZi's Allowlist Referral is an ***allowlist*** and it's confusing to place it under "DNS blocklists" when the UI has a section for "DNS allowlist" -- does it mean the intention of the list is reverted, block the allowlist or is it truly an allowlist?

The UI for "DNS allowlist" should work the same as for blocklists, the user could pick from a list of _presets_ (such as HaGeZi's Allowlist Referral) or add their custom.

https://adguardteam.github.io/HostlistsRegistry/assets/filter_45.txt

<img width="390" alt="image" src="https://github.com/AdguardTeam/AdGuardHome/assets/23507658/5f59111d-8191-45ce-87fe-068d322be26a">

FWIW, this list is _very_ useful to have to not break shopping sites and referral links.

**Comment by ainar-g** [comment]
There doesn't seem to be a separate category for allowlists in the HostlistsRegistry repo, so you should probably propose the change there first, as AdGuard Home merely uses the structures provided there.

**Comment by timkgh** [comment]
> There doesn't seem to be a separate category for allowlists in the HostlistsRegistry repo, so you should probably propose the change there first, as AdGuard Home merely uses the structures provided there.

Not sure what this means, sorry.

**Comment by ainar-g** [comment]
https://github.com/AdguardTeam/HostlistsRegistry

This is the repository with the list of filters that AdGuard Home uses.  All changes in categorization must be proposed there, not here.

**Comment by tyler-abovedesign** [comment]
It's not a change in categorization, it's a change in AG Ux. 
Why doesn't AG give you the option to choose from lists when you are in the "allow lists" settings?

**Comment by kimdre** [comment]
It's quite confusing to me. Are the DNS blocklists and allowlists pages just a separation on the UI level but work exactly the same under the hood? When I try to add HaGeZi's Allowlist Referral to the allowlists while it's also on the blocklists, I get a duplicate list error, so it seems like to me.
Then why are there even two separate filterlist pages when they are pretty much the same?

**Comment by Sil3ntVip3r** [comment]
Current-master inspection confirms this is not fully resolved by
HostlistsRegistry#342.

- The generated catalog still emits `filter_45` as ordinary category `other`.
- The blocklist modal exposes the entire catalog.
- The allowlist modal supports manual name/URL entry only and has no catalog
  picker.
- The backend intentionally creates separate blocklist and allowlist engines,
  so the pages are semantically different.
- URL uniqueness is global across both collections, explaining the reported
  `url already exists` response when attempting to move the same URL to the
  allowlist.

HostlistsRegistry#342 was closed because the registry did not want a separate
category at that time, but the later UI request here remains valid.  The
smallest complete design needs source metadata distinguishing vetted
allowlists, then should exclude those entries from the blocklist picker and
expose them in the allowlist picker.  This issue should therefore remain open
and be classified as both UI work and external metadata work, rather than
treated as resolved solely by #342.

**Comment by gvillo** [comment]
yes, this is not ok, DNS allowlist should have his own pick list + custom
